tests/path.rs

```diff
@@ -0,0 +1,69 @@
+#![cfg_attr(
+    feature = "dev",
+    allow(dead_code, unused_variables, unused_imports, unreachable_code)
+)]
+#![deny(clippy::all)]
+
+#[path = "./utils.rs"]
+mod utils;
+
+use check_protocols::R;
+use test_utils::trim_margin;
+use utils::test_run;
+
+#[test]
+fn looks_up_step_executable_in_path() -> R<()> {
+    test_run(
+        r##"
+            |#!/usr/bin/env bash
+            |cp
+        "##,
+        r##"
+            |protocol:
+            |  - cp
+        "##,
+        Ok(()),
+    )?;
+    Ok(())
+}
+
+#[test]
+fn looks_up_unmocked_command_executable_in_path() -> R<()> {
+    test_run(
+        r##"
+            |#!/usr/bin/env bash
+            |ls $(dirname dir/file)
+        "##,
+        r##"
+            |protocols:
+            |  - protocol:
+            |    - ls dir
+            |unmockedCommands:
+            |  - dirname
+        "##,
+        Ok(()),
+    )?;
+    Ok(())
+}
+
+#[test]
+fn shortens_received_executable_to_file_name_when_reporting_step_error() -> R<()> {
+    test_run(
+        r##"
+            |#!/usr/bin/env bash
+            |mv
+        "##,
+        r##"
+            |protocol:
+            |  - cp
+        "##,
+        Err(&trim_margin(
+            "
+                |error:
+                |  expected: cp
+                |  received: mv
+            ",
+        )?),
+    )?;
+    Ok(())
+}
```

Comment:
Maybe we should have a test for what happens when the script executes e.g. `/usr/bin/something`, and the protocol specifies `something`, but the first thing found in the `$PATH` is `/usr/local/bin/something`.

Similarly we probably want a test that makes sure the user *can* specify that a script is supposed to execute `/usr/bin/something`, even though it's shadowed by `/usr/local/bin/something`.

And maybe we also need a test that make sure users can still specify executables that are not in the `$PATH` at all.

Reply:
I added tests for a script that runs an executable that is not the first match in `$PATH` where the protocol expects an executable that is the first match, for a script that runs the first match in path where the protocol expects an executable that is not the first match, and where the script runs an executable that is not in `$PATH` at all.